packages/api/lambdas/verifyProviderSecretsMigration.js

```diff
@@ -0,0 +1,29 @@
+'use strict';
+
+const KMS = require('@cumulus/aws-client/KMS');
+const { dynamodbDocClient } = require('@cumulus/aws-client/services');
+const { isNil } = require('@cumulus/common/util');
+
+const verifyProvider = async (provider) => {
+  if (provider.encrypted === true) {
+    try {
+      await KMS.decryptBase64String(provider.username);
+      await KMS.decryptBase64String(provider.password);
+    } catch (_) {
+      throw new Error(`Provider ${provider.id} credentials could not be decrypted using KMS. Must invoke the providerSecretsMigration Lambda function.`);
+    }
+  } else {
+    if (isNil(provider.username) && isNil(provider.password)) return;
+    throw new Error(`Provider ${provider.id} has plaintext username or password. Must invoke the providerSecretsMigration Lambda function.`);
+  }
+};
+
+const handler = async () => {
+  const scanResponse = await dynamodbDocClient().scan({
+    TableName: process.env.ProvidersTable
+  }).promise();
+
+  await Promise.all(scanResponse.Items.map(verifyProvider));
```

Comment:
Yes. This isn't being used right now, we're just deploying it so that it will be available in future releases. In the release _after_ the one that includes this migration, we are going to invoke this Lambda as part of the Terraform deployment. It will serve as a circuit breaker to make sure that someone can't deploy a later version of Cumulus without having invoked the migration lambda.